swift/ios-shoppe-demo/ServiceManagers/FullStoryManager.swift

```diff
@@ -97,3 +99,63 @@ func fsModifyPrivacy(setting: PrivacySetting, views: UIView?...) {
 func fsIdentify(id: String, userInfo: [String: Any]) {
     FS.identify(id, userVars: userInfo)
 }
+
+/**
+Generate a unique alphanumeric ID with specified len
+    - Parameter len: Desired length of ID string
+*/
+func makeId(len: Int) -> String {
+    let characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+    var randomString = ""
+    for _ in 0 ..< len {
+        let randomCharacter = characters.randomElement()!
+        randomString.append(randomCharacter)
+    }
+
+    return randomString
+}
+
+// ------------------------
+// below are functions for demostrating conversions: revenue attribution events
+// ------------------------
+func fsAddOrRemoveProductEvent(event: Event, with product: Product?) {
+    guard let product = product else {
+        return
+    }
+    // TODO: check for event type
```

Comment:
These feels a bit loose. But for now, these functions will take in any `event`, we should probably check and make sure they are making sense with the function name before proceeding.